robot-core/src/main/java/org/obolibrary/robot/ExportOperation.java

```diff
@@ -0,0 +1,946 @@
+   *
+   * @param classes Set of class expressions to convert to string
+   * @param provider QuotedAnnotationValueShortFormProvider for Manchester expressions
+   * @param excludeAnonymous if true, exclude anonymous class expressions
+   * @return String of class expressions or null
+   */
+  private static String classExpressionsToString(
+      Collection<OWLClassExpression> classes,
+      QuotedAnnotationValueShortFormProvider provider,
+      boolean excludeAnonymous) {
+    List<String> strings = new ArrayList<>();
+    for (OWLClassExpression expr : classes) {
+      if (expr.isAnonymous() && !excludeAnonymous) {
+        // Get a Manchester string using labels
+        String manString = renderManchester(expr, provider);
+        strings.add(manString);
+      } else if (!expr.isAnonymous()) {
+        OWLClass sc = expr.asOWLClass();
+        strings.add(provider.getShortForm(sc));
+      }
+    }
+    if (strings.isEmpty()) {
+      return "";
+    }
+    List<String> sorted =
+        strings.stream().sorted(String::compareToIgnoreCase).collect(Collectors.toList());
+    return String.join("|", sorted);
+  }
+
+  /**
+   * Given an ontology, a quoted entity checker, an entity, a list of columns, a label function, and
+   * a boolean indicating to exclude anonymous classes, return a map of cell values for this entity.
+   *
+   * @param ontology OWLOntology to get details from
+   * @param ioHelper IOHelper to resolve CURIEs
+   * @param checker QuotedEntityChecker to resolve entities
+   * @param e OWLEntity to get details of
+   * @param columns List of columns indicating the details to get
+   * @param provider QuotedAnnotationValueShortFormProvider for Manchester expressions
+   * @param excludeAnonymous if true, exclude anonymous class expressions
+   * @return Map of column names to cell values
+   * @throws Exception if a property (from columns) cannot be resolved
+   */
+  private static Map<String, String> getCellValues(
+      OWLOntology ontology,
+      IOHelper ioHelper,
+      QuotedEntityChecker checker,
+      OWLEntity e,
+      List<String> columns,
+      QuotedAnnotationValueShortFormProvider provider,
+      boolean excludeAnonymous)
+      throws Exception {
+    Map<String, String> cellMap = new HashMap<>();
+    Map<String, IRI> labelMap = OntologyHelper.getLabelIRIs(ontology);
+    for (String col : columns) {
+      // Each column is a label, CURIE, or IRI
+      // First check for the special keywords
+      switch (col) {
+        case "IRI":
+          cellMap.put("IRI", e.getIRI().toString());
+          continue;
+        case "CURIE":
+          cellMap.put("CURIE", e.getIRI().getShortForm().replace("_", ":"));
+          continue;
+        case "LABEL":
+          cellMap.put("LABEL", provider.getShortForm(e));
+          continue;
+        default:
+          break;
+      }
+
+      // Try to resolve a CURIE
+      IRI iri = ioHelper.createIRI(col);
```

Comment:
I think this is figuring out the column header IRI over-and-over for each row. We should do that once for the table.